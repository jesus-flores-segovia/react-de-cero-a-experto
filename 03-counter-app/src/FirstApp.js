import PropTypes from "prop-types";

// Podemos definir valores por defecto al recibir props vacias
const FirstApp = ({greetings= "There is not greetings prop", greetings2, subtitle, subtitle2, author}) => {

    // Tambien podemos realizar cualquier validaci√≥n por tal de asignar un valor a la prop
    //subtitle = subtitle === undefined ? "There is not subtitle prop": null;

    return (
        <>
	        <h1>{greetings}</h1>
            {/*<h1>{greetings2}</h1>*/}
	        <p>{subtitle}</p>
            {/*<p>{subtitle2}</p>*/}
            {/*<pre>{JSON.stringify(author, null, 3)}</pre>*/}
        </>
    );
};

// Mediante PropTypes, es posible configurar validaciones para el uso del componente
FirstApp.propTypes = {
    //greetings2: PropTypes.string.isRequired
};

FirstApp.defaultProps = {
    subtitle2: "Here there is a default subtitle",
    subtitle: "Here there is a default subtitle"
};

export default FirstApp;