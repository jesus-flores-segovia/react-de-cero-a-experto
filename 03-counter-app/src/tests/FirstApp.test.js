import React from "react";
import FirstApp from "../FirstApp";
import {shallow} from "enzyme";
import "@testing-library/jest-dom";

describe("Tests inside file FirstApp.test.js", () =>{
    /* test("'FirstApp' component must show the prop greetings 'Hello, I'm Jesús Flores'", () =>{

        const greetings = "Hello, I'm Jesús Flores";
        const {getByText} = render(<FirstApp greetings={greetings} greetings2={greetings}></FirstApp>);
        console.log(getByText);
        console.log(getByText(greetings));
        //console.log(getByText("Invented"));

        expect(getByText(greetings)).toBeInTheDocument();
    }); */

    test("'FirstApp' component must show correctly", () =>{

        const greetings = "Hello, I'm Jesús Flores";
        const wrapper = shallow(<FirstApp greetings={greetings}></FirstApp>);

        expect(wrapper).toMatchSnapshot();
    });

    test("'FirstApp' component must show the subtitle passed by the prop value", () =>{

        const greetings = "Hello, I'm Jesús Flores";
        const subtitle = "This is a default subtitle";
        const wrapper = shallow(<FirstApp greetings={greetings} subtitle={subtitle}></FirstApp>);

        const subtitleText = wrapper.find("p").text();

        expect(subtitleText).toBe(subtitle);
    });
});