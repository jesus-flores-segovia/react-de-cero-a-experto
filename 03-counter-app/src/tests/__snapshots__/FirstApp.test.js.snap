// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests inside file FirstApp.test.js 'FirstApp' component must show correctly 1`] = `
<Fragment>
  <h1>
    Hello, I'm Jes√∫s Flores
  </h1>
  <p>
    Here there is a default subtitle
  </p>
</Fragment>
`;
